### ISSUE TITLE
Unrecognized subroutine

### ISSUE THREAD

**Issue by sponce24** [cmacmackin/ford#140]
Hello, 

Thank you for creating this software ! Its great ! 

I have a small issue. If subroutines are call within a subroutine, they are not 
detected in the graphs produced by Ford. 
If the called subroutine is in the same .f90 file, then it works. 

For example, say I have two source file and three subroutine:
file1.f90 and file2.f90
Say file1.f90 contains:
```
subroutine file1
  CALL file2
end subroutine file1
```
and file2.f90 contains:
```
subroutine file2
  CALL test
end subroutine file2
subroutine test
 write(*,*) test
end subroutine test
```
Then the links created for the subroutine file1 will not work but the call to test in file2 will work. 

Both *.f90 files are in the same /src directory. 
How do I make the dependence so the links are correct ?

Thank you, 

Samuel

**Comment by cmacmackin** [comment]
This was a weakness in how I programmed FORD to look for matching 
procedures. It doesn't look for procedures in other files unless they 
are used via a module. I'll come up with a way around this shortcoming 
for the next release.

On 31/05/16 19:39, sponce24 wrote:
>
> Hello,
>
> Thank you for creating this software ! Its great !
>
> I have a small issue. If subroutines are call within a subroutine, 
> they are not
> detected in the graphs produced by Ford.
> If the called subroutine is in the same .f90 file, then it works.
>
> For example, say I have two source file and three subroutine:
> file1.f90 and file2.f90
> Say file1.f90 contains:
>
> |subroutine file1 CALL file2 end subroutine file1 |
>
> and file2.f90 contains:
>
> |subroutine file2 CALL test end subroutine file2 subroutine test 
> write(*,*) test end subroutine test |
>
> Then the links created for the subroutine file1 will not work but the 
> call to test in file2 will work.
>
> Both *.f90 files are in the same /src directory.
> How do I make the dependence so the links are correct ?
>
> Thank you,
>
> Samuel
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cmacmackin/ford/issues/140>, or mute the thread 
> <https://github.com/notifications/unsubscribe/AHxJPXBx6ZTXKtCdfuqOv9Pu1Qzk9L45ks5qHIBYgaJpZM4Iq1JG>.
>

-- 
Chris MacMackin
cmacmackin.github.io <http://cmacmackin.github.io>

**Comment by sponce24** [comment]
Hi Chris, 

Thank you for your reply. 
The software we develop is actually part of a larger solid state physics software package.
Last time we met, we discussed about adding automatic documentation and Ford was mention (along with others). After fiddling a bit with them I think Ford best suits our need. I would now need to convince the other developers. 

Would you be able to give me a rough time frame for the following two implementations?:
- pdf generation
- detection of subroutines that are not in a module

Thank you again for this great software !

PS: I'm fully aware that this is your hobby and not your job and therefore timescale might be long.  

Best, 

Samuel

**Comment by cmacmackin** [comment]
It's difficult to say. Right now is quite a busy time for me with my 
studies. Getting the non-module subroutines could (conceivably) be done 
within a month or so. PDF output is a much bigger project and I hesitate 
to give a timeline. Most likely I won't get to start it until mid-July. 
It's difficult to know how long it would then take to complete. Assuming 
nothing went too horribly wrong, I would probably have it working by the 
fall. However, there are plenty of other of feature requests which I 
have lined up as well, some of which have been waiting much longer than 
this, so any timeline I give will be very rough.

On 01/06/16 10:55, sponce24 wrote:
>
> Hi Chris,
>
> Thank you for your reply.
> The software we develop is actually part of a larger solid state 
> physics software package.
> Last time we met, we discussed about adding automatic documentation 
> and Ford was mention (along with others). After fiddling a bit with 
> them I think Ford best suits our need. I would now need to convince 
> the other developers.
>
> Would you be able to give me a rough time frame for the following two 
> implementations?:
>
>   * pdf generation
>   * detection of subroutines that are not in a module
>
> Thank you again for this great software !
>
> PS: I'm fully aware that this is your hobby and not your job and 
> therefore timescale might be long.
>
> Best,
>
> Samuel
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cmacmackin/ford/issues/140#issuecomment-222946859>, or 
> mute the thread 
> <https://github.com/notifications/unsubscribe/AHxJPXssHD2AbtV4Phtn6uNImG6aLK3mks5qHVckgaJpZM4Iq1JG>.
>

-- 
Chris MacMackin
cmacmackin.github.io <http://cmacmackin.github.io>

**Comment by zbeekman** [comment]
@sponce24 you can always try to implement these features yourself if time is critical... I'd imagine Chris would be willing to consider merging your improvements back upstream, so long as they are implemented in a sane manner...

**Comment by sponce24** [comment]
@cmacmackin Thank you for the timescale. This is great and the pdf feature is less important (in my opinion) anyhow.
I just wanted to say that the MathJax issue fix (online/off line) works perfectly in 4.6.2 ! Thank you.

**Comment by cmacmackin** [comment]
I've gotten the issue with unrecognized procedures fixed. It will be in the next release, probably out in a week or two. I'm closing this issue now; although PDF documentation is mentioned here and is not yet implemented, it wasn't the purpose of this issue.
